packages/block-editor/src/components/global-styles/color-panel.js

```diff
@@ -0,0 +1,754 @@
+				},
+				{
+					key: 'hover',
+					label: __( 'Hover' ),
+					inheritedValue: hoverLinkColor,
+					setValue: setHoverLinkColor,
+					userValue: userHoverLinkColor,
+				},
+			],
+		},
+	].filter( Boolean );
+
+	elements.forEach( ( { name, label, showPanel } ) => {
+		if ( ! showPanel ) return;
+
+		const elementBackgroundColor = decodeValue(
+			inheritedValue?.elements?.[ name ]?.color?.background
+		);
+		const elementGradient = decodeValue(
+			inheritedValue?.elements?.[ name ]?.color?.gradient
+		);
+		const elementTextColor = decodeValue(
+			inheritedValue?.elements?.[ name ]?.color?.text
+		);
+		const elementBackgroundUserColor = decodeValue(
+			value?.elements?.[ name ]?.color?.background
+		);
+		const elementGradientUserColor = decodeValue(
+			value?.elements?.[ name ]?.color?.gradient
+		);
+		const elementTextUserColor = decodeValue(
+			value?.elements?.[ name ]?.color?.text
+		);
+		const hasElement = () =>
+			!! (
+				elementTextUserColor ||
+				elementBackgroundUserColor ||
+				elementGradientUserColor
+			);
+		const resetElement = () => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							background: undefined,
+							gradient: undefined,
+							text: undefined,
+						},
+					},
+				},
+			} );
+		};
+
+		const setElementTextColor = ( newValue ) => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							text: encodeColorValue( newValue ),
+						},
+					},
+				},
+			} );
+		};
+		const setElementBackgroundColor = ( newValue ) => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							background: encodeColorValue( newValue ),
+							gradient: undefined,
+						},
+					},
+				},
+			} );
+		};
+		const setElementGradient = ( newValue ) => {
+			onChange( {
+				...value,
+				elements: {
+					...value?.elements,
+					[ name ]: {
+						...value?.elements?.[ name ],
+						color: {
+							...value?.elements?.[ name ]?.color,
+							gradient: encodeGradientValue( newValue ),
+							background: undefined,
+						},
+					},
+				},
+			} );
+		};
+		const supportsTextColor = true;
+		const supportsBackground = name !== 'caption';
+
+		items.push( {
+			key: name,
+			label,
+			hasValue: hasElement,
+			resetValue: resetElement,
+			isShownByDefault: defaultControls[ name ],
+			indicators:
+				supportsTextColor && supportsBackground
+					? [
+							elementTextColor,
+							elementBackgroundColor ?? elementGradient,
+					  ]
+					: [
+							supportsTextColor
+								? elementTextColor
+								: elementBackgroundColor ?? elementGradient,
```

Comment:
Same as above, should these be flipped so that the gradient is preferenced before the background color?